sdk/nodejs/npm/workspaces_test.go

```diff
@@ -45,3 +45,22 @@ func TestFindWorkspaceRootYarnExtended(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, filepath.Join("testdata", "workspace-extended"), root)
 }
+
+func TestFindWorkspaceRootNested(t *testing.T) {
```

Comment:
Is it worth adding a couple of similar tests to this in typescript as well?

Reply:
Definitely, turns out I didn't add any tests for the Typescript version, added in https://github.com/pulumi/pulumi/pull/15601/commits/86e07431e49b9016a6596b18aba399dce90a9384